### PR TITLE
Enable cinder with ceph backend

### PR DESCRIPTION
**Pull request by jovial**

### CODE REVIEW

**Comment by jovial** [comment]
kolla ansible only deploys `cinder_volume` on hosts in the storage group. Maybe we should change
https://github.com/openstack/kayobe/blob/master/ansible/roles/kolla-ansible/templates/overcloud-services.j2#L106?

File: etc/kayobe/kolla/globals.yml [hunk 1]
```diff
@@ -2,9 +2,10 @@
 # Additional global kolla-ansible configuration.
 
 # Enable Ceph as the backend for glance image storage.
+# FIXME: Use file backend for now.
```
[the remark this answers is not in the document]
Comment (jovial) [reply etc/kayobe/kolla/globals.yml]: Does that relate to the nova backend as well?

**Comment by markgoddard** [comment]
It's possible to override various parts of the kolla inventory. See https://github.com/SKA-ScienceDataProcessor/alaska-kayobe-config/blob/alaska-prod/etc/kayobe/kolla/inventory/overcloud-components.j2 for an example. We would need to override the overcloud-service.j2 file, you can find the defaults in https://github.com/openstack/kayobe/blob/master/ansible/roles/kolla-ansible/templates/overcloud-services.j2.

**Comment by jovial** [comment]
This seems to work now though:
```
(openstack) stack@eod-seed:~/will/kayobe-env/src/kayobe$ openstack volume create --image CentOS7.5 --size 10 bootable_volume                                                                                                                                                     
+---------------------+--------------------------------------+
| Field               | Value                                |
+---------------------+--------------------------------------+
| attachments         | []                                   |
| availability_zone   | nova                                 |
| bootable            | false                                |
| consistencygroup_id | None                                 |
| created_at          | 2018-06-20T16:16:46.000000           |
| description         | None                                 |
| encrypted           | False                                |
| id                  | 0a62392e-e087-4ebd-b3a6-2564077f09e0 |
| migration_status    | None                                 |
| multiattach         | False                                |
| name                | bootable_volume                      |
| properties          |                                      |
| replication_status  | None                                 |
| size                | 10                                   |
| snapshot_id         | None                                 |
| source_volid        | None                                 |
| status              | creating                             |
| type                | None                                 |
| updated_at          | None                                 |
| user_id             | aa63d0eb27bd41fbb8a7de48556a972e     |

(openstack) stack@eod-seed:~/will/kayobe-env/src/kayobe$ openstack volume list
+--------------------------------------+-----------------+-------------+------+-------------+
| ID                                   | Name            | Status      | Size | Attached to |
+--------------------------------------+-----------------+-------------+------+-------------+
| 0a62392e-e087-4ebd-b3a6-2564077f09e0 | bootable_volume | downloading |   10 |             |
+--------------------------------------+-----------------+-------------+------+-------------+
(openstack) stack@eod-seed:~/will/kayobe-env/src/kayobe$ openstack volume list
+--------------------------------------+-----------------+-----------+------+-------------+
| ID                                   | Name            | Status    | Size | Attached to |
+--------------------------------------+-----------------+-----------+------+-------------+
| 0a62392e-e087-4ebd-b3a6-2564077f09e0 | bootable_volume | available |   10 |             |
+--------------------------------------+-----------------+-----------+------+-------------+
(openstack) stack@eod-seed:~/will/kayobe-env/src/kayobe$ openstack volume show bootable_volume
```

snippet from `openstack volume show`:

```
| name                           | bootable_volume                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                 
                                  |
| os-vol-host-attr:host          | rbd:volumes@rbd-1#rbd-1   
```

File: etc/kayobe/kolla/globals.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (markgoddard) [reply etc/kayobe/kolla/globals.yml]: It was just for glance

File: etc/kayobe/kolla/globals.yml [hunk 2]
```diff
@@ -2,9 +2,10 @@
...
 glance_backend_ceph: "yes"
-cinder_backend_ceph: "no"
-nova_backend_ceph: "no"
+cinder_backend_ceph: "yes"
+#nova_backend_ceph: "no"
```
Comment (markgoddard) [inline etc/kayobe/kolla/globals.yml]: I think we still want this

File: etc/kayobe/kolla/config/cinder/cinder-volume.conf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[DEFAULT]
+enabled_backends=rbd-1
```
[the remark this answers is not in the document]
Comment (jovial) [reply etc/kayobe/kolla/config/cinder/cinder-volume.conf]: done

**Comment by jovial** [comment]
I think we are good now:
```
(openstack) 2 stack@eod-seed:~/will/kayobe-env/src/kayobe$ openstack volume show bootable_volume2 -c os-vol-host-attr:host                                                                                                                                                       +-----------------------+-----------------------+
| Field                 | Value                 |
+-----------------------+-----------------------+
| os-vol-host-attr:host | rbd:volumes@ceph#ceph |
+-----------------------+-----------------------+
```

**Comment by markgoddard** [comment]
Can you boot an instance using a volume?

**Comment by jovial** [comment]
I shall check...

**Pull request merged**